scripts/stackstorm.js

```diff
@@ -171,7 +174,11 @@ module.exports = function(robot) {
       .catch(function (err) {
         robot.logger.error('Failed to authenticate: ' + err.message);
 
-        throw err;
+        if (env.EXIT_ON_FAILURES) {
+          stop({exit: true});
```

Comment:
What if another error is thrown in different place of the code? Will the hubot-stackstorm wait in the same noop state?

If we reached the state when the raw error is thrown to user, - that was an automatic exit before (expected & correct).

I suspect something changed in recent refactorings or NodeJS upgrades?
Some handler was broken?

Reply:
I was curious about this, so I dug into it. This changed [recently](https://github.com/hubotio/hubot/blob/d2b57bd796749fbb9ca02eff33b7a345bcd40d68/docs/scripting.md#error-handling):

> Previously, an uncaught exceptions would crash your hubot instance. Hubot now includes an uncaughtException handler, which provides hooks for scripts to do something about exceptions.

Hubot catches uncaught exceptions itself now. Previously it would just crash.

Reply:
Awesome, so that's the root cause.

Sounds like normally we would just need to rely on that handler and add exit with non-zero code without changing any other logic.

@blag Thanks for finding it! :+1:

Reply:
@jinpingh please pay attention to this ^^ discussion.

Try adding a handler that'll exit on every error/exception instead of patching specific cases and leaving any other possible failure scenarios buggy as reported in StackStorm/st2chatops#124.

Reply:
@armab Current fixes take care of all st2 related configuration issues.  For `hubot` configuration issue (for examples: With invalid HUBOT_SLACK_TOKEN) it raises exception from `hubot` and  `st2chatops` is still hanging.

Reply:
OK, I think something related to the discussion happening in this thread + research + was the right path to solve this issue.
We've talked before about global handler that will catch similar unrecoverable issues. I think https://github.com/StackStorm/hubot-stackstorm/pull/172 is the simpler way to solve the root cause.

Reply:
As noted, #172 is not a proper fix for this issue, and parts of this PR have been reverted to more closely match previous behavior.